subsys/net/ip/igmp.c

```diff
@@ -326,3 +330,34 @@ int net_ipv4_igmp_leave(struct net_if *iface, const struct in_addr *addr)
 					sizeof(struct in_addr));
 	return ret;
 }
+
+void net_ipv4_igmp_init(struct net_if *iface)
+{
+	struct net_if_mcast_addr *maddr;
```

Comment:
We could check here that there are multicast addresses defined (unlikely but good to check). So something like
```
if (CONFIG_NET_IF_MCAST_IPV4_ADDR_COUNT < 1) {
   return;
}
```

Reply:
Thanks, I've incorporated this into my latest change.